Sming/Arch/Esp8266/Components/esp8266/startup.cpp

```diff
@@ -106,7 +106,7 @@ extern "C" void ICACHE_FLASH_ATTR WEAK_ATTR user_pre_init(void)
 	static const partition_item_t partitions[] = {
 			PARTITION_ITEM(SYSTEM_PARTITION_BOOTLOADER,			0,					1),
 			PARTITION_ITEM(SYSTEM_PARTITION_RBOOT_CONFIG,		1,					1),
-			PARTITION_ITEM(SYSTEM_PARTITION_PROGRAM,			2,					MAX_PROGRAM_SECTORS - 2),
+			PARTITION_ITEM(SYSTEM_PARTITION_PROGRAM,			2,					(rfCalSector < MAX_PROGRAM_SECTORS ? rfCalSector : MAX_PROGRAM_SECTORS) - 2),
```

Comment:
Revert this change. PR #1800 will take care of it.

Reply:
Done.